src/external/visualization_integration.cu

```diff
@@ -52,11 +56,46 @@ __host__ bool send_particles_to_visualization(const std::string &url, MapNode *n
             ",\"y\":" + vector_double_to_json_array(y) +
             ",\"z\":" + vector_double_to_json_array(z) + "}";
 
-    http::Request request(url);
+
+    http::Request particles_request(urls.first);
+
+    try
+    {
+        const http::Response response = particles_request.send("POST", body, {"Content-Type: application/json"});
+
+        if(response.status < 200 || 300 <= response.status)
+            throw http::ResponseError("Response status is not OK");
+    }
+    catch(const std::exception &e)
+    {
+        std::cerr << "Request failed, error: " << e.what() << std::endl;
+        return false;
+    }
+
+
+
+    for(int i = 0; i < n_of_faces; ++i)
+    {
+        for(int j = 0; j < polyhedron->get_faces()[i].get_n_of_vertices(); ++j)
+        {
+            SpacePoint v = polyhedron->get_faces()[i].get_vertices()[j];
+            poly.push_back({v.x, v.y, v.z});
+        }
+    }
+
+    body = "[ ";
+    for(int i = 0; i < poly.size(); ++i)
+    {
+        body += vector_double_to_json_array(poly[i]);
+        if (i != poly.size() - 1) { body += ", "; }
```

Comment:
1. I don't think we want to add spaces after commas because we want to save the traffic.
2. This is not serious and I wouldn't mention it unless there were another issue in this piece of code (the above one), but I guess, this would look better as two lines:
```suggestion
        if (i != poly.size() - 1)
            body += ',';
```